src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -660,6 +675,9 @@ private void UpdateRenderTreeToMatchClientState(ulong eventHandlerId, EventField
         /// <param name="disposing"><see langword="true"/> if this method is being invoked by <see cref="IDisposable.Dispose"/>, otherwise <see langword="false"/>.</param>
         protected virtual void Dispose(bool disposing)
         {
+            // It's important that we handle all exceptions here before reporting any of them.
+            // This way we can dispose all components before an error handler kicks in.
+            List<Exception> exceptions = null;
```

Comment:
Unlike the above, I do see why it's necessary to `try`/`catch` here. Definitely looks like a bug that we didn't before.

Reply:
This has the same requirements though. You may need to dispose multiple components when the renderer is disposed. You may also need to dispose multiple components as part of the single batch.